File: src/main/java/exnihilocreatio/registries/types/FluidBlockTransformer.java
package exnihilocreatio.registries.types;

import exnihilocreatio.util.BlockInfo;
import exnihilocreatio.util.EntityInfo;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import net.minecraft.item.crafting.Ingredient;

@AllArgsConstructor
@EqualsAndHashCode
public class FluidBlockTransformer {

    @Getter
    private String fluidName;

    @Getter
    private Ingredient input;

    @Getter
    private BlockInfo output;

    @Getter
    private EntityInfo toSpawn;

    @Getter
    private int spawnCount;

    @Getter
    private int spawnRange;

    public FluidBlockTransformer(String fluidName, Ingredient input, BlockInfo output){
        this(fluidName, input, output, (String) null, 0, 0);
    }

    public FluidBlockTransformer(String fluidName, Ingredient input, BlockInfo output, String entityName){
        this(fluidName, input, output, entityName, 4, 4);
    }

    public FluidBlockTransformer(String fluidName, Ingredient input, BlockInfo output, String entityName, int spawnCount, int spawnRange){
        this.fluidName = fluidName;
        this.input = input;
        this.output = output;
        this.toSpawn = entityName == null ? EntityInfo.EMPTY : new EntityInfo(entityName);
        this.spawnCount = spawnCount;
        this.spawnRange = spawnRange;
    }

}

File: src/main/java/exnihilocreatio/registries/types/FluidBlockTransformer.kt
package exnihilocreatio.registries.types

import exnihilocreatio.util.BlockInfo
import exnihilocreatio.util.EntityInfo
import net.minecraft.item.crafting.Ingredient

// TODO: MOVE TO API PACKAGE ON BREAKING VERSION CHANGE
data class FluidBlockTransformer(val fluidName: String,
                                 val input: Ingredient,
                                 val output: BlockInfo,
                                 val toSpawn: EntityInfo = EntityInfo.EMPTY,
                                 val spawnCount: Int = 4,
                                 val spawnRange: Int = 4) {

    @JvmOverloads
    constructor(fluidName: String, input: Ingredient, output: BlockInfo, entityName: String? = null, spawnCount: Int = 4, spawnRange: Int = 4)
            : this(fluidName, input, output, if (entityName == null) EntityInfo.EMPTY else EntityInfo(entityName), spawnCount, spawnRange)

}
